# Import Libraries

In [17]:
import pandas as pd 
pd.options.plotting.backend = "plotly"
import numpy as np

# Load Datasets

In [18]:
data_train = pd.read_parquet("../data/interim/data_train.parquet")

In [19]:
for column in data_train:
    print(f'{column} = {(data_train[column].isnull().sum()/len(data_train))*100}')

Efficiency = 10.35248380375827
BoilTime = 10.310015875594623
Color = 10.330822310198288
BoilGravity = 13.782980336494202
OG = 10.249876729000466
Name = 10.321701681330927
Size(L) = 10.248166611087834
PrimaryTemp = 35.68816569902495
BrewMethod = 10.296619951945686
SugarScale = 10.327117054720922
ABV = 10.307735718377781
FG = 10.340512978369858
IBU = 10.307450698725678
Style = 10.872359649197811


# Description and Data Cleaning

## Variable definitions
- Input/Output variables
- Variable Types (Categorical or Numeric)
- Data Types (int, float, string, factor, boolean, ...)

|Feature Name| Type | Description and values | % Missing
|--|--|--|--|
| Name | Categorical (nominal) | The beer name | 10.30% 
| Size(L)| Number | Amount brewed for recipe listed | 10.23% 
| OG | Number | Specific gravity of wort before fermentation | 10.25% 
| FG | Number (nominal) | Specific gravity of wort after fermentation | 10.30% 
| ABV | Number | Alcohol By Volume | 10.29% 
| IBU | Number | International Bittering Units | 10.28% 
| Color | Number | Standard Reference Method - light to dark ex. 40 = black-  | 10.32% 
| BoilTime | Number | Time wort is boiled | 10.38% 
| BoilGravity | Number | Specific gravity of wort before the boil | 13.76% 
| Efficiency | Number | Beer mash extraction efficiency - extracting sugars from the grain during mash | 10.36% 
| SugarScale | Categorical (nominal) | Scale to determine the concentration of dissolved solids in wort | 10.30%
| BrewMethod | Categorical (nominal) | Various techniques for brewing | 10.24% 
| PrimaryTemp | Number | Temperature at the fermenting stage | 35.63% 
| Style| Categorical (nominal) | The type of brew | 10.87% 



In [20]:
data_train.sample(10)

Efficiency       BoilTime          Color   BoilGravity             OG  \
6479         75.0           90.0  -354128741541        1058.0  -354128741541   
106795       70.0             60          11.81          None           1039   
92401        35.0           None           2.99        1046.0         1053.0   
200444       35.0             45           6.89          1089           1062   
158968       69.0   856842215413   856842215413          1057           1065   
279051       70.0             60          15.67          1072           1099   
314991       70.0  -354128741541          12.15          1059           1081   
77682        75.0             60           17.1  856842215413           1062   
107796       65.0           90.0          42.63        1057.0         1074.0   
350268       70.0           60.0           50.0        1078.0         1085.0   

                              Name Size(L) PrimaryTemp    BrewMethod  \
6479                         St�vt    25.0        17.0     All Grain   
106795                 Scottish 60    None        None          BIAB   
92401         Mosaic Promise Clone   18.93        20.0       extract   
200444                  Mosaic IPA   18.93       17.78       extract   
158968          Dead Clone Guy Ale    40.0        15.5  856842215413   
279051             Titanic Iceberg    20.8        None     All Grain   
314991        Gizmos Golden Galaxy   20.82        20.0     All Grain   
77682              Hoppy LHBS Pale   18.93        None     All Grain   
107796  Foreign Extra Stout - Base   18.93        20.0          BIAB   
350268             Stout La Grande    60.0        20.5     All Grain   

              SugarScale   ABV              FG     IBU                   Style  
6479    Specific Gravity  5.72          1014.0   31.31             Sweet Stout  
106795  Specific Gravity  3.42          1012.0   17.14     Scottish Heavy 70/-  
92401               None   5.4            None   50.27       American Pale Ale  
200444  Specific Gravity  6.59          1012.0   90.86            American IPA  
158968  Specific Gravity  7.24  856842215413.0    37.1     Maibock/Helles Bock  
279051  Specific Gravity  None            None   14.56                 Eisbock  
314991  Specific Gravity  8.08          1019.0  104.18           -354128741541  
77682   Specific Gravity  5.38          1021.0    27.6  California Common Beer  
107796  Specific Gravity  None          1021.0   83.27     Foreign Extra Stout  
350268  Specific Gravity  None          1017.0   53.32          Imperial Stout

In [21]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 350853 entries, 0 to 350852
Data columns (total 14 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   Efficiency   314531 non-null  object
 1   BoilTime     314680 non-null  object
 2   Color        314607 non-null  object
 3   BoilGravity  302495 non-null  object
 4   OG           314891 non-null  object
 5   Name         314639 non-null  object
 6   Size(L)      314897 non-null  object
 7   PrimaryTemp  225640 non-null  object
 8   BrewMethod   314727 non-null  object
 9   SugarScale   314620 non-null  object
 10  ABV          314688 non-null  object
 11  FG           314573 non-null  object
 12  IBU          314689 non-null  object
 13  Style        312707 non-null  object
dtypes: object(14)
memory usage: 37.5+ MB


Null Values

In [22]:
data_train.isnull().sum()

Efficiency      36322
BoilTime        36173
Color           36246
BoilGravity     48358
OG              35962
Name            36214
Size(L)         35956
PrimaryTemp    125213
BrewMethod      36126
SugarScale      36233
ABV             36165
FG              36280
IBU             36164
Style           38146
dtype: int64